api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -315,8 +316,8 @@ public void disableExtension(String extensionId) {
 		return extensions;
 	}
 	
-	// making it public is a hack so we can test this directly in the appui module
-	public boolean checkRequireExpression(Extension candidate, AppContextModel contextModel) {
+	@Override
```

Comment:
Is this intentionally backwards incompatible?

Reply:
Yeah, I was going to suggest keeping the old checkRequireExpression around as well... but currently it's only public as a hack to allow tests to run (there's not checkRequireExpression(Extension,AppContext on the actual AppFrameworkService interface)

Reply:
oh i see. 😊

Reply:
I can leave it around, but anywhere that wound up with this version of `checkRequireExpression()` will also have the version of `Extension` that implements the `Requireable` interface. That and the fact that this wasn't a publicly available method before means I think this has a very low chance of actually breaking anything.

Reply:
i agree.

Reply:
I agree as well!